File: benchmarks/src/nodejs/prime_count.js
function isPrime(n) {
    if (n < 2) return false;
    if (n === 2) return true;
    if (n % 2 === 0) return false;
    for (let i = 3; i * i <= n; i += 2) {
        if (n % i === 0) return false;
    }
    return true;
}

const start = Date.now();
let count = 0;
for (let n = 2; n <= 100000; n++) {
    if (isPrime(n)) count++;
}
const end = Date.now();
console.log(`Result: ${count}`);
console.log(`Time: ${end - start}ms`);

File: benchmarks/src/ruby/prime_count.rb
def is_prime(n)
  return false if n < 2
  return true if n == 2
  return false if n % 2 == 0
  i = 3
  while i * i <= n
    return false if n % i == 0
    i += 2
  end
  true
end

count = 0
start = Time.now.to_f * 1000
(2..100000).each { |n| count += 1 if is_prime(n) }
finish = Time.now.to_f * 1000
puts "Result: #{count}"
puts "Time: #{(finish - start).round}ms"
